### ISSUE TITLE
Errors in the function to optimize

### ISSUE THREAD

**Issue by cryanking** [AnotherSamWilson/ParBayesianOptimization#9]
I don't think errors in the function to be optimized are really handled. The argument "errorHandling" is not passed to foreach for the initial grid. Any grid elements that error result in inconsistent numbers of columns in the rbind .combine argument to foreach. Any grid elements that return NA cause errors in addIterations, where the functions used to determine scale aren't NA-safe. 

This comes up in some problems where the FUN can sporadically fail in internal optimization (despite best efforts). I am building in restarts, jittering (for continuous params), and subsampling to try an ensure calculations succeed, but some parameter values can make optimization difficult. One could also imagine isolated singularities in the parameters that one didn't know about to "protect". Assigning a "bad" score as a work-around seems to confuse the GP part of the algorithm.

```
test_fun_NA <- function(x) {
  if(x %% 4 == 0) {
    return(NA)
  } else {
    dnorm(x,3,2)*1.5 + dnorm(x,7,1) + dnorm(x,10,2)
  }
}

test_fun_err <- function(x) {
  if(x %% 4 == 0) {
    stop("compute err")
  } else {
    dnorm(x,3,2)*1.5 + dnorm(x,7,1) + dnorm(x,10,2)
  }
}


bounds <- list(x=c(0L,15L))
initGrid <- data.frame(x=c(0L,2L,4L, 5L, 7L, 8L, 9L, 10L))

FUN <- function(x) list(Score = test_fun_err(x))

set.seed(6)
optObjSimp <- ParBayesianOptimization::bayesOpt(
  FUN = FUN
  , bounds = bounds
  , initGrid = initGrid
  , iters.n = 2
, errorHandling = "continue"
)


FUN <- function(x) list(Score = test_fun_NA(x))

set.seed(6)
optObjSimp <- ParBayesianOptimization::bayesOpt(
  FUN = FUN
  , bounds = bounds
  , initGrid = initGrid
  , iters.n = 2
, errorHandling = "continue"
)

```

**Comment by AnotherSamWilson** [comment]
You're right - unit tests only include a 2 input function for error handling, should be simple enough to get bayesOpt to handle errors from a 1 input function. I didn't include protections around the return of NAs, but that should be pretty simple also. Should be done today.

**Comment by AnotherSamWilson** [comment]
Fixed in 1.2.0
